### DISCUSSION TITLE
OpenDAL 内部实现：数据读取

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#409] [Announcements]
# OpenDAL 内部实现：数据读取

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/2023/02-how-opendal-read-data/

**Comment by microbubble9586** [comment]
> 这个 body 内部包含的 stream 是 reqwest 返回的 bytes stream，opendal 在此基础上实现了 content length 检查和 read 支持。

捉虫 reqwest -> request

**Reply by Xuanwo** [reply]
`bytes_stream` 是 `reqwest` crate 库中的一个方法，返回了一个 `impl Stream`

**Reply by microbubble9586** [reply]
对不起，是我丈育了🥲🥲，默认当成request了

**Reply by Xuanwo** [reply]
没有 context 可能确实比较容易看错，我改进一下这里的描述～
